### PR TITLE
MoveAction capability can drop cancel request if it is sent shortly after goal is sent

### PR DESCRIPTION
**Pull request by khupilz**
### Description

Preempt request could be dropped in `MoveGroupMoveAction` capability if it is sent shortly after goal is sent. 

The problem is that the `preemptMoveCallback` sets the `PlanExecution::preempt_requested_` as `true` but `executeMoveCallback_PlanAndExecute` resets the `PlanExecution::preempt_requested_` as `false`. If `preemptMoveCallback` is called before `plan_execution_->planAndExecute`, the preempt request is  lost. In this case the goal status is set to `PREEMPTING` but the goal is planned and executed. In the end a result with error code `SUCCESS` is received. A test is provided to reproduce the problem.

An easy fix is provided by adding a similar `preempt_requested_` flag in  `MoveGroupMoveAction` and checking this flag before `plan_execution_->planAndExecute` is called. At the end of `executeMoveCallback` the flag is reset to `false`. 

I am not sure if this is the right place to fix since there could be also similar problems in other action capabilities. 

### Checklist
- [x] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [x] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/move_group/package.xml [hunk 1]
```diff
@@ -32,6 +32,10 @@
   <run_depend>pluginlib</run_depend>
   <run_depend>std_srvs</run_depend>
 
+  <test_depend>rostest</test_depend>
+  <test_depend>rosunit</test_depend>
```
Comment (rhaschke) [inline moveit_ros/move_group/package.xml]: Are you sure `rosunit` is required? Looks like you are using only standard python `unittest` and `rostest`.

Comment (khupilz) [reply moveit_ros/move_group/package.xml]: Hi Robert, you are right. `rosunit` is needless. I removed it.

**Comment by rhaschke** [comment]
@khupilz Sometimes, the unittest still fails:
https://travis-ci.org/ros-planning/moveit/jobs/341626697#L1098

Could you have another look?

**Comment by rhaschke** [comment]
@khupilz: Ping.

**Pull request merged**